docs/contributing-to-airbyte/building-new-connector/standard-source-tests.md

```diff
@@ -0,0 +1,53 @@
+# Standard Source Test Suite
+Test methods start with `test`. Other methods are internal helpers in the java class implementing the test suite.
+### getImageName
+Name of the docker image that the tests will run against.
+
+### getSpec
+Specification for integration. Will be passed to integration where appropriate in each test. Should be valid.
+
+### getConfig
+Configuration specific to the integration. Will be passed to integration where appropriate in each test. Should be valid.
+
+### getConfiguredCatalog
+The catalog to use to validate the output of read operations. This will be used as follows: Full Refresh syncs will be tested on all the input streams which support it Incremental syncs: - if the stream declares a source-defined cursor, it will be tested with an incremental sync using the default cursor. - if the stream requires a user-defined cursor, it will be tested with the input cursor in both cases, the input getState() will be used as the input state.
+
+### getState
+No method description was provided
+
+### getRegexTests
+List of regular expressions that should match the output of the test sync.
+
+### setup
+Function that performs any setup of external resources required for the test. e.g. instantiate a postgres database. This function will be called before EACH test.
+
+### tearDown
+Function that performs any clean up of external resources required for the test. e.g. delete a postgres database. This function will be called after EACH test. It MUST remove all data in the destination so that there is no contamination across tests.
+
+### setUpInternal
+No method description was provided
+
+### tearDownInternal
+No method description was provided
+
+### testGetSpec
```

Comment:
why is a person coming to this page? are they trying to understand what tests are run and why they could fail? or is it something more? I am wondering if it would be better to just include the methods that are actually tests and not all of the public methods. depends on why you think someone is coming to this page.

Reply:
i thought the user story was, "i am _not_ a java developer, i just had a test fail and i wanna know why if failed and i don't want to read a bunch of java". if that's the case maybe it's better to just show the test methods?

Reply:
great question. let me see if I can do that easily. I was worried about filtering by method name (in case a test doesn't start with `test*`), but maybe I could filter via the `@Test` annotation.

Reply:
Done!